include/operations/StandardOperation.hpp

```diff
@@ -72,6 +72,26 @@ class StandardOperation : public Operation {
 
   [[nodiscard]] bool isStandardOperation() const override { return true; }
 
+  void addControls(const Controls& c) override {
+    for (auto ctrl : c) {
+      if (actsOn(ctrl.qubit)) {
+        throw QFRException(
+            "Cannot add control to operation as it already acts on "
+            "the control qubit.");
+      }
+
+      controls.insert(ctrl);
+    }
+  }
+
+  void clearControls() override { controls.clear(); }
+
+  void removeControls(const Controls& c) override {
+    for (auto ctrl : c) {
+      controls.erase(ctrl);
+    }
+  }
```

Comment:
This method should probably also contain a sanity check that is the inverse of the `addControl` case, i.e., it should throw an exception if the gate does not act on the qubit to be removed. Even more specific, the qubit to be removed should be an existing control.